packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,40 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+const coreStep: CoreStepFunction = async function ({ buildDir, netlifyConfig }) {
+  const configPath = resolve(buildDir, '.netlify', 'deploy', 'v1', 'config.json')
+
+  let config: Partial<NetlifyConfig> = {}
+
+  try {
+    const data = await fs.readFile(configPath, 'utf8')
+
+    config = JSON.parse(data) as Partial<NetlifyConfig>
+  } catch {
+    return {}
+  }
+
+  const newConfig = merge(netlifyConfig, config)
```

Comment:
What rules does this use for things like redirect arrays? Is the order defined?

Reply:
Right now it's merging them so that the entries from the deploy configuration come last, which is what we want in redirects, headers, and edge functions. We'll need special handling once/if we want to implement the `headers!` notation we discussed for defining precedence over user-defined values.

Reply:
Sounds good to me